modules/ref_fb_module/src/statistics_fb_impl.cpp

```diff
@@ -310,6 +351,63 @@ void StatisticsFbImpl::processDataPacket(const DataPacketPtr& packet)
     domainSignal.sendPacket(outDomainPacket);
 }
 
+void StatisticsFbImpl::processDataPacketInput(const DataPacketPtr& packet)
+{
+    if (!valid)
+        return;
+
+    const auto domainPacket = packet.getDomainPacket();
+    if (!domainPacket.assigned())
+        return;
+
+    if (triggerMode)
+    {
+        const auto packetBuf = static_cast<uint8_t*>(packet.getData());
+        auto domainBuf = static_cast<Int*>(domainPacket.getData());
```

Comment:
There's a lot of assumptions about the data types of the domain and trigger signal here. I'd still create a short function `validateTriggerDescriptors` that's called when a `DataDescriptorChanged` packet is received on the trigger input port.